src/App.tsx

```diff
@@ -8,52 +8,45 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
 import { Post } from './types/Post';
-import { Counter } from './features/counter/Counter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import {
+  getPostsAction,
+  selectedPostAction,
+} from './features/posts/postsSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const author = useAppSelector(state => state.users.selectUser);
+  const hasError = useAppSelector(state => state.posts.error);
+  const posts = useAppSelector(state => state.posts.listPosts);
+  const loaded = useAppSelector(state => state.posts.loader);
```

Comment:
would be better to create some file like `selectors.ts`

and export 
```
export const selectAuthor = (state: RootState) => state.users.selectUser;
export const selectPostError = (state: RootState) => state.posts.error


usage:
const author = useAppSelector(selectAuthor);
const hasError = useAppSelector(selectPostError);
```

we need to do it, if path to some value change, you'll need to change it in only 1 place

Reply:
I recommend you resolve this comment, cause its a better solution.

Reply:
it would be better to move these selectors